### PR TITLE
feat(rnd) Add Postgres support

### PR DESCRIPTION
**Pull request by aarushik93**
### Background

Adding support for Postgres, to enable cloud support. 

### Changes 🏗️
Added a prisma config for postgres
Added migrations
Installed Postgres in CI for ubuntu, mac, windows
Removed build steps in CI as they will not work with Postgres and are no longer needed.

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [x] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

___

### **PR Type**
Enhancement, Configuration changes, Documentation

___

### **Description**
- Added support for PostgreSQL including Prisma configuration and migrations.
- Updated CI workflow to install and configure PostgreSQL.
- Added example environment variables for PostgreSQL configuration.
- Added Docker Compose configuration for PostgreSQL.
- Removed unnecessary build steps in CI workflow.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>6 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>db.py</strong><dd><code>Load environment variables and set Prisma schema path</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/data/db.py

<li>Added dotenv loading to fetch environment variables.<br> <li> Set <code>PRISMA_SCHEMA_PATH</code> environment variable.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7513/files#diff-041613dfd4db749045023bfb1d08e6ca9b9aefdbfe385e49f3cdb5692c42d8ce">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>migration.sql</strong><dd><code>Add SQLite migration script</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/migrations/20240722143555_migrations/migration.sql

- Added SQL migration script for SQLite.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7513/files#diff-45eb4ed8c2c4a9417b424aae9031f787fc916cad47d57be17a8267e7677407b2">+94/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>migration_lock.toml</strong><dd><code>Add SQLite migration lock file</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/migrations/migration_lock.toml

- Added migration lock file for SQLite.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7513/files#diff-d35f5ebd27edbbcfa9fd7f90ed49d9482178980799b3df24f5d9c3e53b6e45af">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>migration.sql</strong><dd><code>Add PostgreSQL migration script</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/migrations_postgres/20240722143307_migrations/migration.sql

- Added SQL migration script for PostgreSQL.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7513/files#diff-edfc79f970822f49731cf55edbc2fb15a8731801a7ac056e14e34b1ff2bb6541">+128/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>migration_lock.toml</strong><dd><code>Add PostgreSQL migration lock file</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/migrations_postgres/migration_lock.toml

- Added migration lock file for PostgreSQL.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7513/files#diff-ac03d0e5a5974b339bc258d06132bb0290c791d8443ef958ceed0d0ed18e7c83">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>schema_postgres.prisma</strong><dd><code>Add Prisma schema for PostgreSQL</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/schema_postgres.prisma

- Added Prisma schema for PostgreSQL.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7513/files#diff-ac19ce19501048edff1348d799b96ddf8d3414cce0132ae6895ab7ce26717e53">+148/-0</a>&nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>autogpt-server-ci.yml</strong><dd><code>Update CI workflow to support PostgreSQL</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/autogpt-server-ci.yml

<li>Added setup for PostgreSQL in CI workflow.<br> <li> Removed build steps in CI workflow.<br> <li> Updated environment variables for database connection.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7513/files#diff-be2c4106e439333623427dccd03a5208c025493d5d67025bf67c23134c2058bd">+22/-150</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>docker-compose.yml</strong><dd><code>Add Docker Compose configuration for PostgreSQL</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/docker-compose.yml

- Added Docker Compose configuration for PostgreSQL.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7513/files#diff-aa81ced61fb669ab634a2fe62b6731081ac5635ae2dd1019f0900d1c932e297c">+16/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>.env.example</strong><dd><code>Add example environment variables for PostgreSQL</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/.env.example

- Added example environment variables for PostgreSQL configuration.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7513/files#diff-9dd1e3bfac62caa79a5a78c43140da9ea45d0d3f4d20f14b1fe894d6c40ce0af">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Update README formatting</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/README.md

- Minor formatting changes.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7513/files#diff-c70604dfb8498ee90b025bbe6a7be3e996510f2308bd01d2e96bfeb27c04a933">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>pyproject.toml</strong><dd><code>Add python-dotenv dependency</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/pyproject.toml

- Added `python-dotenv` dependency.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7513/files#diff-adb694b1ef2fb9dabbdd0efe486d08bc9336bcd7b824f13f4125228d70cd4d7a">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by aarushik93** [comment]
Based on the discussions on discord regarding DB engines and ORMs, I'm making this a draft PR so it doesn't accidentally get merged

File: rnd/autogpt_server/migrations/migration_lock.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Please do not edit this file manually
```
Comment (aarushik93) [inline rnd/autogpt_server/migrations/migration_lock.toml:1]: Added the SQLite migrations, given we are dropping installer support right now.

File: rnd/autogpt_server/README.md [hunk 2]
```diff
@@ -49,8 +49,8 @@ We use the Poetry to manage the dependencies. To set up the project, follow thes
 
    ```sh
    poetry run prisma migrate dev
-   ```
-   
+  ```
+
```
Comment (Swiftyos) [inline rnd/autogpt_server/README.md:53]: Can you update the README to include details on `PRISMA_SCHEMA_PATH` please

**Review by Swiftyos — CHANGES_REQUESTED** [review]

**Review by Swiftyos — APPROVED** [review]

**Review by ntindle — APPROVED** [review]
Would be cool to check in ci if there’s the same number of migrations for both

**Review by ntindle — APPROVED** [review]

**Pull request merged**